connect/runtime/src/main/java/org/apache/kafka/connect/util/KafkaBasedLog.java

```diff
@@ -119,22 +143,27 @@ public void start() {
         consumer = createConsumer();
 
         List<TopicPartition> partitions = new ArrayList<>();
+        TopicDescription topicDesc = topicSupplier.getOrCreateTopic(topic);
+        if (topicDesc != null) {
+            for (Map.Entry<Integer, TopicPartitionInfo> entry : topicDesc.partitions().entrySet()) {
```

Comment:
I didn't put any debug messages in `KafkaBasedLog` because at this point it's not clear whether the topics were in fact created or whether they had already existed. Instead, the `TopicAdmin` utility that does all the work has log statements to report whether for every topic whether it was created or found to already exist.

However, I do think that it's worth adding to this code a similar log message to what's in `TopicAdmin` when it falls back to auto-creation. That happens starting on line 148 and is used only when the `TopicAdmin` can't use the `AdminClient` with the given broker -- should ideally never happen but is a potential condition.